paimon-core/src/main/java/org/apache/paimon/table/source/DataTableStreamScan.java

```diff
@@ -177,6 +182,10 @@ private Plan nextPlan() {
                 throw new EndOfScanException();
             }
 
+            if (checkDelaySnapshot(nextSnapshotId)) {
+                continue;
```

Comment:
There will be a busy loop?

Can we just return `SnapshotNotExistPlan.INSTANCE` here?